### PR TITLE
Use python logging

### PR DESCRIPTION
**Pull request by ssbarnea**
Instead of using print use logging module for messages that
are not direct result of the tool (linting error).

### CODE REVIEW

File: lib/ansiblelint/utils.py [hunk 1]
```diff
@@ -52,6 +53,26 @@
...
+
+logger = logging.getLogger()
+handler = logging.StreamHandler()
+formatter = logging.Formatter('%(levelname)-8s %(message)s')
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+logger.setLevel(logging.DEBUG)
+
+
+def logger_verbosity(level):
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:73]: It should be a verb
```suggestion
def set_logger_verbosity(level):
```

Comment (webknjaz) [reply lib/ansiblelint/utils.py:73]: Also plz add a docstring

File: lib/ansiblelint/utils.py [hunk 2]
```diff
@@ -382,7 +403,8 @@ def normalize_task_v2(task):
             task_pprint = pp.pformat(task)
         except ImportError:
             task_pprint = task
-        raise SystemExit("Couldn't parse task at %s (%s)\n%s" % (task_info, e.message, task_pprint))
+        logger.fatal("Couldn't parse task at %s (%s)\n%s" % (task_info, e.message, task_pprint))
+        raise SystemExit(ANSIBLE_FAILURE_RC)
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:407]: Changing the return code conventions is out of the scope of this PR.

File: test/TestCliRolePaths.py [hunk 3]
```diff
@@ -20,16 +20,14 @@ def run_ansible_lint(self, cwd, bin, role_path, env=None):
             env=env,
         ).communicate()
 
-        self.assertFalse(err, 'Expected no error but was ' + str(err))
-
-        return result
+        return result, stderr
```
Comment (webknjaz) [inline test/TestCliRolePaths.py:23]: I don't see any point in returning stderr, you're not using it
```suggestion
        return result
```

File: test/TestCliRolePaths.py [hunk 4]
```diff
@@ -10,7 +10,7 @@ def setUp(self):
     def run_ansible_lint(self, cwd, bin, role_path, env=None):
         command = '{} {}'.format(bin, role_path)
 
-        result, err = subprocess.Popen(
+        result, stderr = subprocess.Popen(
```
Comment (webknjaz) [inline test/TestCliRolePaths.py:13]: this looks unrelated

File: lib/ansiblelint/utils.py [hunk 5]
```diff
@@ -52,6 +53,26 @@
...
+INVALID_CONFIG_RC = 2
+ANSIBLE_FAILURE_RC = 3
+
+VERBOSITY_MAP = {
+    0: logging.NOTSET,
+    1: logging.INFO,
+    2: logging.DEBUG
+}
+
+logger = logging.getLogger()
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:65]: Not sure if I want only one global logger to be used. Why didn't you go in the direction of per-module loggers?

Comment (ssbarnea) [reply lib/ansiblelint/utils.py]: Yep, better even if ansiblelint is a cli tool.

File: lib/ansiblelint/__init__.py [hunk 6]
```diff
@@ -19,12 +19,12 @@
 # THE SOFTWARE.
 
 from collections import defaultdict
+import codecs
```
Comment (webknjaz) [inline lib/ansiblelint/__init__.py:22]: (future hint) it's probably unnecessary now that we don't have to support Python 2. Out of scope of this PR, of course, but would be a nice improvement separately.

File: lib/ansiblelint/cli.py [hunk 7]
```diff
@@ -51,11 +52,10 @@ def load_config(config_file):
 
     if config_file:
         if not os.path.exists(config_path):
-            print(
-                "Config file not found '{cfg!s}'.".format(cfg=config_path),
-                file=sys.stderr,
+            logger.info(
+                "Config file not found '{cfg!s}'.".format(cfg=config_path)
```
Comment (webknjaz) [inline lib/ansiblelint/cli.py:56]: Linters (pylint?) usually yell at manual interpolation in logger calls, it's preferable to pass arguments separately. Would something like this work?
```suggestion
                "Config file not found '%(cfg)s'.", cfg=config_path)
```

File: lib/ansiblelint/cli.py [hunk 8]
```diff
@@ -51,11 +52,10 @@ def load_config(config_file):
 
     if config_file:
         if not os.path.exists(config_path):
-            print(
-                "Config file not found '{cfg!s}'.".format(cfg=config_path),
-                file=sys.stderr,
+            logger.info(
```
Comment (webknjaz) [inline lib/ansiblelint/cli.py:55]: Why is this one info but the one below is error?

File: lib/ansiblelint/generate_docs.py [hunk 9]
```diff
@@ -45,7 +46,7 @@ def main():
     filename = '../../docs/docsite/rst/rules/default_rules.rst'
     with open(filename, 'w') as file:
         file.write(make_table(grid))
-        print('{} file written'.format(filename))
+        logger.info('{} file written'.format(filename))
```
Comment (webknjaz) [inline lib/ansiblelint/generate_docs.py:49]:
```suggestion
        logger.info('%s file written', filename)
```

File: lib/ansiblelint/utils.py (hunk 5 above)
Comment (webknjaz) [reply lib/ansiblelint/utils.py:65]: Could you please elaborate on this? Not sure I can parse this.

File: lib/ansiblelint/utils.py (hunk 1 above)
Comment (webknjaz) [reply lib/ansiblelint/utils.py:73]: @ssbarnea ^ functions perform actions, so their names should hint that some action is done by using a verb. Otherwise, it reads as a property/variable.

File: test/TestCliRolePaths.py [hunk 10]
```diff
@@ -20,8 +20,6 @@ def run_ansible_lint(self, cwd, bin, role_path, env=None):
             env=env,
         ).communicate()
 
-        self.assertFalse(err, 'Expected no error but was ' + str(err))
```
Comment (webknjaz) [inline test/TestCliRolePaths.py:23]: Why? Can this be changed to use a test using a `caplog` fixture?

File: lib/ansiblelint/utils.py [hunk 11]
```diff
@@ -857,12 +877,10 @@ def get_playbooks_and_roles(options=None):
             playbooks.append(normpath(p))
             continue
 
-        if options.verbosity:
-            print('Unknown file type: %s' % normpath(p))
+        logger.info('Unknown file type: %s' % normpath(p))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:880]:
```suggestion
        logger.info('Unknown file type: %s', normpath(p))
```

File: lib/ansiblelint/utils.py [hunk 12]
```diff
@@ -857,12 +877,10 @@ def get_playbooks_and_roles(options=None):
...
 
-    if options.verbosity:
-        print('Found roles: ' + ' '.join(role_dirs))
-        print('Found playbooks: ' + ' '.join(playbooks))
+    logger.info('Found roles: ' + ' '.join(role_dirs))
+    logger.info('Found playbooks: ' + ' '.join(playbooks))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:883]:
```suggestion
    logger.info('Found roles: %s', ' '.join(role_dirs))
    logger.info('Found playbooks: %s', ' '.join(playbooks))
```

File: lib/ansiblelint/utils.py [hunk 13]
```diff
@@ -785,13 +806,12 @@ def get_yaml_files(options):
...
-                "Warning: Failed to discover yaml files to lint using git:"
-                " {error_msg}".format(
-                    error_msg=exc.output.rstrip('\n')
-                )
+        logger.warning(
+            "Failed to discover yaml files to lint using git:"
+            " {error_msg}".format(
+                error_msg=exc.output.rstrip('\n')
             )
+        )
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:814]:
```suggestion
        logger.warning(
            "Failed to discover yaml files to lint using git:"
            " %(error_msg)s",
            error_msg=exc.output.rstrip('\n'),
        )
```

File: lib/ansiblelint/utils.py [hunk 14]
```diff
@@ -771,12 +793,11 @@ def get_yaml_files(options):
...
-    if options.verbosity:
-        print(
-            "Discovering files to lint: {command}".format(
-                command=' '.join(git_command)
-            )
+    logger.info(
+        "Discovering files to lint: {command}".format(
+            command=' '.join(git_command)
         )
+    )
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:800]:
```suggestion
    logger.info(
        "Discovering files to lint: %(command)s",
        command=' '.join(git_command),
    )
```

File: lib/ansiblelint/utils.py [hunk 15]
```diff
@@ -600,7 +622,7 @@ def append_skipped_rules(pyyaml_data, file_text, file_type):
         yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
     except RuntimeError as exc:
         # Notify user of skip error, do not stop, do not change exit code
-        print('Error trying to append skipped rules: {!r}'.format(exc))
+        logger.error('Error trying to append skipped rules: {!r}'.format(exc))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:625]:
```suggestion
        logger.error('Error trying to append skipped rules: %r', exc)
```

Comment (webknjaz) [reply lib/ansiblelint/utils.py:625]: Even better
```suggestion
        logger.exception('Error trying to append skipped rules')
```

Comment (webknjaz) [inline lib/ansiblelint/utils.py:625]: `exc` is not really needed
```suggestion
    except RuntimeError:
        # Notify user of skip error, do not stop, do not change exit code
        logger.exception('Error trying to append skipped rules')
```

File: lib/ansiblelint/utils.py [hunk 16]
```diff
@@ -382,7 +403,8 @@ def normalize_task_v2(task):
             task_pprint = pp.pformat(task)
         except ImportError:
             task_pprint = task
-        raise SystemExit("Couldn't parse task at %s (%s)\n%s" % (task_info, e.message, task_pprint))
+        logger.fatal("Couldn't parse task at %s (%s)\n%s" % (task_info, e.message, task_pprint))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:406]: There's no `fatal` here: https://docs.python.org/3/library/logging.html. Did you mean
```suggestion
        logger.critical("Couldn't parse task at %s (%s)\n%s", task_info, e.message, task_pprint)
```

File: lib/ansiblelint/cli.py (hunk 7 above)
Comment (ssbarnea) [reply lib/ansiblelint/cli.py:56]: Clearly it would work. I did not touch it because use of new formatter was your suggestion last week. Another alternative is to use one of the hackish solutions that add formatting support to logging (a custom formatter). Still, i am bit afraid that we may endup introducing more problems than solving. Better to go to older but fully supported syntax.

File: test/TestCliRolePaths.py (hunk 10 above)
Comment (ssbarnea) [reply test/TestCliRolePaths.py:23]: The reason for removing this line is that now we get some valid output on stderr. Presence of output on stderr no longer means an error happened.

At this moment I do not know what kind of test we can reliably do about stderr, it can be empty, or have various content even on perfect valid executions.

**Comment by ssbarnea** [comment]
Current failure seems to be CI issue, which can be avoided if we merge https://github.com/ansible/ansible-lint/pull/737/files first.

File: lib/ansiblelint/cli.py (hunk 7 above)
Comment (webknjaz) [reply lib/ansiblelint/cli.py]: Oh, let me make it clear then. Formatting of the usual strings should be done with the use of `str.format()` (and f-strings in the future) but it's a bit different for logging: when you pass a separate pre-rendered string it doesn't know about the vars but when you pass them separately it does. I think the idea of separating the args comes from structlog.

Also, pylint has [logging-not-lazy (W1201) rule](http://pylint.pycqa.org/en/latest/technical_reference/features.html#logging-checker-messages) that implies that lazy interpolation is preferable (meaning that it won't happen if not really necessary per logging config).

File: test/TestCliRolePaths.py (hunk 10 above)
Comment (webknjaz) [reply test/TestCliRolePaths.py:23]: I noticed that you changed the return code too. So maybe you could assert that? Or how about using `subprocess.check_output()` instead?

**Comment by webknjaz** [comment]
@ssbarnea I just realized that I was commenting on the outdated chain (https://github.com/ansible/ansible-lint/pull/732#discussion_r410730458) so let me just copy this here for discoverability:

Formatting of the usual strings should be done with the use of `str.format()` (and f-strings in the future) but it's a bit different for logging: when you pass a separate pre-rendered string it doesn't know about the vars but when you pass them separately it does. I think the idea of separating the args comes from structlog.

Besides, tools that aggregate logs (like Sentry) can group them properly: https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html#comment-2063573645.

Also, pylint has [logging-not-lazy (W1201) rule](http://pylint.pycqa.org/en/latest/technical_reference/features.html#logging-checker-messages) that implies that lazy interpolation is preferable (meaning that it won't happen if not really necessary per logging config).

The [official logging doc](https://docs.python.org/3/howto/logging.html#optimization) also recommends lazy interpolation: https://stackoverflow.com/a/34634301/595220.

File: lib/ansiblelint/utils.py [hunk 17]
```diff
@@ -52,6 +53,27 @@
...
+    1: logging.INFO,
+    2: logging.DEBUG
+}
+
+logger = logging.getLogger("ansiblelint")
+handler = logging.StreamHandler()
+formatter = logging.Formatter('%(levelname)-8s %(message)s')
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+logger.setLevel(logging.DEBUG)
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:70]: I think it's not a good idea to have this on the module level globally. Import-time evaluation is usually not a good thing. I'd prefer a setup function called when the app starts.

**Comment by webknjaz** [comment]
@ssbarnea it looks like you forgot to answer the question https://github.com/ansible/ansible-lint/pull/732#discussion_r407121559:
Why do you prefer one global logger vs. multiple loggers (one per module)?

File: lib/ansiblelint/utils.py (hunk 17 above)
Comment (ssbarnea) [reply lib/ansiblelint/utils.py:70]: @webknjaz Can you please give an existing logging implementation as an reference example so I can  follow that? 

I understand what you mean and I agree it may not be the best implementation but I also want to avoid rewriting this PR forever. For example, I would have avoided implementing set_logging_verbosity_level and create an init_logging() function instead if that remark was added earlier.

Another approach is to merge it in its current form and implement the other changes as a follow-up, as the logging calls would not change.

File: test/TestCliRolePaths.py (hunk 10 above)
Comment (ssbarnea) [reply test/TestCliRolePaths.py:23]: @webknjaz This method has no knowledge of what is the outcome of the execution and is called 7 times in tests. Doing any change on how we validate the tests should count as outside the scope of implementing the use of python logging.

**Comment by ssbarnea** [comment]
> @ssbarnea it looks like you forgot to answer the question [#732 (comment)](https://github.com/ansible/ansible-lint/pull/732#discussion_r407121559):
> Why do you prefer one global logger vs. multiple loggers (one per module)?

Latest implementation uses a single logger named `ansiblelint` for the entire package. There is not need to use different loggers as it would make controlling of logging harder.

File: lib/ansiblelint/utils.py (hunk 17 above)
Comment (webknjaz) [reply lib/ansiblelint/utils.py:70]: Basically, I want some `initialize_logger()` implemented here and `main()` calling it at the very beginning, before anything else. I understand that you want to merge incrementally so I won't block this PR if you postpone this specific change.

File: test/TestCliRolePaths.py (hunk 10 above)
Comment (webknjaz) [reply test/TestCliRolePaths.py:23]: Okay

**Comment by webknjaz** [comment]
> Latest implementation uses a single logger named `ansiblelint` for the entire package. There is not need to use different loggers as it would make controlling of logging harder.

I understand this is you opinion and my question is: why do you think so? This looks like a result of some chain of thoughts, it'd be great if you could write them down here.

I have a slightly different opinion:
1) Using a global logger that's always imported from utils creates more inter-module dependencies, can lead to potential circular imports and it's unpredictable when the logger gets initialized.
2) One common practice is to use something like
   ```python
   _logger = logging.getLogger()
   ```
   This method
   1. Makes it easier to trace messages back to the modules that produced them
   2. Allows us to not depend on utils and just initialize loggers where they are needed w/o having to maintain a singleton.
3) Python's stdlib `logging` is done in a few stages:
   ```
   (1) add a log message -> (2) filter by levels -> (3) output if needed
   ```
   We should configure (2) and (3) globally and not care what logger the log messages are coming from. It'd allow modules to have logger instances that are not tightly bound to global output and filtering configuration.

**Comment by ssbarnea** [comment]
As I said, I do not care very much about initial implementation, but I care not to spend endless hours reworking it and failing to switch print with logging calls. As far we get rid of print() I am more than happy.

In fact many people are just calling the module functions directly without an instance, which avoids the singleton issue.

So before I rewrite it, **are you ok if I just use the logging module directly?**
```
import logging
logging.warning(...)
```

**Comment by webknjaz** [comment]
I'd rather not do that. Logger instances should be created per module, inputs should be distinctive, while the output may be aggregated.
Different loggers will have different names and it's easier to figure out where messages came from when you have the emitter identifiers.

File: lib/ansiblelint/utils.py [hunk 18]
```diff
@@ -781,13 +799,10 @@ def get_yaml_files(options):
...
-            print(
-                "Warning: Failed to discover yaml files to lint using git:"
-                " {error_msg}".format(
-                    error_msg=exc.output.rstrip('\n')
-                )
-            )
+        logger.warning(
+            "Failed to discover yaml files to lint using git: %s",
+            exc.output.rstrip('\n')
+        )
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:805]: Note: #738 adds a new print statement below this place, be sure to replace it if it's merged first.

Comment (webknjaz) [reply lib/ansiblelint/utils.py:805]: @ssbarnea it's merged. And besides adding an extra except block, it also has tests that check stdout so you'll need to adjust them as well.

File: test/TestUtils.py [hunk 19]
```diff
@@ -160,23 +161,34 @@ def test_expand_paths_vars(monkeypatch):
...
+        'Discovering files to lint: git ls-files *.yaml *.yml')
+
+    assert expected_info in caplog.record_tuples
+    for msg in caplog.messages:
+        if msg.startswith(message_prefix):
+            return
+    pytest.fail('%s not found in logged messages: %s', msg, caplog.messages)
 
 
 @pytest.mark.parametrize(
```
Comment (webknjaz) [inline test/TestUtils.py:194]: @ssbarnea you missed `capsys` here too:

File: test/TestUtils.py [hunk 20]
```diff
@@ -160,23 +161,34 @@ def test_expand_paths_vars(monkeypatch):
...
+    expected_info = (
+        "ansiblelint",
+        logging.INFO,
+        'Discovering files to lint: git ls-files *.yaml *.yml')
+
+    assert expected_info in caplog.record_tuples
+    for msg in caplog.messages:
+        if msg.startswith(message_prefix):
+            return
+    pytest.fail('%s not found in logged messages: %s', msg, caplog.messages)
```
Comment (webknjaz) [inline test/TestUtils.py:191]: It's better to avoid manual `pytest.fail()` and also it reads better semantically if you do something like "`assert` there's `any` message in captured `messages` that `startswith` `message_prefix`":
```suggestion
    assert any(m.startswith(message_prefix) for m in caplog.messages)
```

File: test/TestUtils.py [hunk 21]
```diff
@@ -160,23 +161,34 @@ def test_expand_paths_vars(monkeypatch):
...
 )
 def test_get_yaml_files_git_verbose(
     reset_env_var,
     message_prefix,
     monkeypatch,
-    capsys
+    caplog
 ):
     options = cli.get_config(['-v'])
+    utils.initialize_logger(options.verbosity)
```
Comment (webknjaz) [inline test/TestUtils.py:178]: I think this should be put into some sort of a generator-based fixture that would reset the logging level on teardown.

File: lib/ansiblelint/utils.py [hunk 22]
```diff
@@ -863,12 +878,10 @@ def get_playbooks_and_roles(options=None):
             playbooks.append(normpath(p))
             continue
 
-        if options.verbosity:
-            print('Unknown file type: %s' % normpath(p))
+        _logger.info('Unknown file type: %s', normpath(p))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:881]: It looks like you're changing the logic here. It's probably a warning

File: lib/ansiblelint/utils.py [hunk 23]
```diff
@@ -863,12 +878,10 @@ def get_playbooks_and_roles(options=None):
...
 
-    if options.verbosity:
-        print('Found roles: ' + ' '.join(role_dirs))
-        print('Found playbooks: ' + ' '.join(playbooks))
+    _logger.info('Found roles: %s', ' '.join(role_dirs))
+    _logger.info('Found playbooks: %s', ' '.join(playbooks))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:884]: should this be debug?

File: lib/ansiblelint/utils.py [hunk 24]
```diff
@@ -52,6 +53,29 @@
 
 PLAYBOOK_DIR = os.environ.get('ANSIBLE_PLAYBOOK_DIR', None)
 
+INVALID_CONFIG_RC = 2
+ANSIBLE_FAILURE_RC = 3
+
+_logger = logging.getLogger("ansiblelint")
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:59]:
```suggestion
_logger = logging.getLogger(__name__)
```

File: lib/ansiblelint/utils.py (hunk 22 above)
Comment (ssbarnea) [reply lib/ansiblelint/utils.py:881]: I do not think so because the logging will happen or not based on verbosity level, is just that now we rely on logging to make the decision.

File: lib/ansiblelint/utils.py [hunk 25]
```diff
@@ -52,6 +53,29 @@
...
+    }
+
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter('%(levelname)-8s %(message)s')
+    handler.setFormatter(formatter)
+    _logger.addHandler(handler)
+    _logger.setLevel(logging.DEBUG)
+
+    if level is not None:
+        _logger.setLevel(VERBOSITY_MAP.get(level, logging.DEBUG))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:77]: this is essentially
```suggestion
    _logger.setLevel(VERBOSITY_MAP.get(level, logging.DEBUG))
```

File: lib/ansiblelint/__init__.py [hunk 26]
```diff
@@ -278,10 +280,10 @@ def run(self):
         # remove files that have already been checked
         files = [x for x in files if x['path'] not in self.checked_files]
         for file in files:
-            if self.verbosity > 0:
-                print("Examining %s of type %s" % (
-                    ansiblelint.utils.normpath(file['path']),
-                    file['type']))
+            _logger.info(
```
Comment (webknjaz) [inline lib/ansiblelint/__init__.py:283]: maybe
```suggestion
            _logger.debug(
```

File: lib/ansiblelint/utils.py (hunk 23 above)
Comment (ssbarnea) [reply lib/ansiblelint/utils.py:884]: Debug will not be displayed unless extra verbosity is asked and this commands are far too important to not be enabled by default (as we had several bugs around collection failures, like false-positives or false-negatives). I think is desirable to keep them like this, at least for the moment. In the long run, we may downgrade them to debug level. I would wait for a user to raise it as a request first, and later make a mini-survey.

File: lib/ansiblelint/utils.py (hunk 25 above)
Comment (ssbarnea) [reply lib/ansiblelint/utils.py]: done

File: lib/ansiblelint/utils.py (hunk 24 above)
Comment (ssbarnea) [reply lib/ansiblelint/utils.py]: done

File: test/TestUtils.py (hunk 20 above)
Comment (ssbarnea) [reply test/TestUtils.py]: done

File: test/TestUtils.py (hunk 19 above)
Comment (ssbarnea) [reply test/TestUtils.py:194]: In fact I did not miss it, i removed it as it was not used anymore.

Comment (webknjaz) [reply test/TestUtils.py:191]: No, you removed it in the test above, it's a different test that still has it. (`test_get_yaml_files_silent`)

File: test/TestUtils.py (hunk 21 above)
Comment (ssbarnea) [reply test/TestUtils.py:178]: Read https://docs.pytest.org/en/latest/logging.html#caplog-fixture  which states that caplog fixture takes care of restoring logging setup after each test run.

File: test/TestUtils.py (hunk 19 above)
Comment (ssbarnea) [reply test/TestUtils.py:191]: In that case there was a bug, wrongly named variable. We still want to check that no output is produced on stderr. The code was functional but it used misleading variable name. I fixed it and also documented the purpose of the test.

File: test/TestUtils.py [hunk 27]
```diff
@@ -185,6 +194,10 @@ def test_get_yaml_files_git_verbose(
     ids=('in Git', 'outside Git'),
 )
 def test_get_yaml_files_silent(is_in_git, monkeypatch, capsys):
+    """Verify that no stderr output is displayed while discovering yaml files.
```
Comment (webknjaz) [inline test/TestUtils.py:197]:
```suggestion
    """Verify that no stderr output is displayed while discovering yaml files.

    (when the verbosity is off, regardless of the Git or Git-repo presence)
```

File: test/TestUtils.py [hunk 28]
```diff
@@ -185,6 +194,12 @@ def test_get_yaml_files_git_verbose(
     ids=('in Git', 'outside Git'),
 )
 def test_get_yaml_files_silent(is_in_git, monkeypatch, capsys):
```
Comment (webknjaz) [inline test/TestUtils.py:196]: You should also check it with `caplog` because originally this test was added to check that there's no warnings when falling back to pure-python lookup.

Comment (ssbarnea) [reply test/TestUtils.py:196]: I am inclined to believe that using capsys is more comprehensive than caplog, mainly because what caplog captures is a subset of what capsys does, usually they are the same but there is no guarantee that someone does not slip a `print()` by mistake.

Anyway, I do not have any strong feelings about the test so I would likely accept any edit made on it.
